### PR TITLE
fmrivols2conn now accepts nibabel nifti objects as input both for fmri image and atlas

### PR DESCRIPTION
**Pull request by vinferrer**
## Proposed changes
- Now ``fmri_input`` allows four types of inputs:
--``str`` for one fmri image path
-- list of ``str`` for different fmri files
-- nifti nibabel object
-- list of nifti nibabel object

### CODE REVIEW

File: brainspace/utils/volflow.py [hunk 1]
```diff
@@ -1,19 +1,23 @@
 from nilearn import datasets
 from nilearn.input_data import NiftiLabelsMasker
-from nilearn.connectome import ConnectivityMeasure 
+from nilearn.connectome import ConnectivityMeasure
+import nibabel as nib
 import numpy as np
 
 
-def fmrivols2conn(fmri_filenames, atlas_filename, confounds_fn="", measure='correlation'):
+def fmrivols2conn(fmri_input, atlas_filename, confounds_fn="", measure='correlation'):
```
Comment (PeerHerholz) [inline brainspace/utils/volflow.py:8]: should `confounds_fn=""` be changed to `confounds=None`?

File: brainspace/utils/volflow.py [hunk 2]
```diff
@@ -23,24 +27,24 @@ def fmrivols2conn(fmri_filenames, atlas_filename, confounds_fn="", measure='corr
...
-            timeseries.append(masker.fit_transform(volume, confounds=confounds_fn[i]).T)
-    timeseries = np.array(timeseries)
-    mean_ts = np.mean(timeseries,axis=0)
-    # call fit_transform from ConnectivityMeasure object
-    FC_matrix = connectome_measure.fit_transform([mean_ts.T])[0]
+    if isinstance(fmri_input,list):
+        timeseries = []
+        # loop that extracts the timeseries for each volume
+        for i,volume in enumerate(fmri_input):
+            if confounds_fn == "":
```
Comment (PeerHerholz) [inline brainspace/utils/volflow.py:39]: related to the question above: `if confounds is None:`

File: brainspace/utils/volflow.py (hunk 1 above)
Comment (llevitis) [reply brainspace/utils/volflow.py:8]: agreed w @PeerHerholz re confounds being set to None instead of ""

**Comment by vinferrer** [comment]
> Thanks a lot @vinferrer, LGTM! I added two small questions. One general thing I would like to ask: should the `confounds` as is or should the code be adapted to utilize [`load_confounds`](https://github.com/SIMEXP/fmriprep_load_confounds) as in [tutorial_0](https://brainspace.readthedocs.io/en/latest/python_doc/auto_examples/plot_tutorial0.html#sphx-glr-python-doc-auto-examples-plot-tutorial0-py) to make it a bit more consistent? @ReinderVosDeWael

I have been looking to the Nilearn documentation and i think the masker.fit_transform already accepts arrays as inputs:
confounds: list of confounds, optional

    List of confounds (2D arrays or filenames pointing to CSV files). Must be of same length than imgs_list.
https://nilearn.github.io/modules/generated/nilearn.input_data.NiftiMasker.html

**Comment by vinferrer** [comment]
I have tested the function and now with the last commit confounds can be an array when only one subject is inputed to the function. Taking that into account now confounds_fn can be a single np array or a list of numpy arrays if more than a subject is introduced

**Comment by vinferrer** [comment]
Should we merge then?

**Pull request merged**